src/features/projectsV2/components/ProjectSnippet.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ReactElement, useContext, useMemo } from 'react';
+import {
+  ConservationProjectConcise,
+  ConservationProjectExtended,
+  TreeProjectConcise,
+  TreeProjectExtended,
+} from '@planet-sdk/common';
+import { useRouter } from 'next/router';
+import { useTranslations } from 'next-intl';
+import { ParamsContext } from '../../common/Layout/QueryParamsContext';
+import ProjectInfoSection from './microComponents/ProjectInfoSection';
+import ImageSection from './microComponents/ImageSection';
+import style from '../styles/ProjectSnippet.module.scss';
+
+interface Props {
+  project:
+    | TreeProjectConcise
+    | ConservationProjectConcise
+    | TreeProjectExtended
+    | ConservationProjectExtended;
+  shouldDisplayPopup: boolean;
+}
+
+export interface CommonProps {
+  slug: string;
+  isApproved: boolean;
+  isTopProject: boolean;
+  allowDonations: boolean;
+  purpose: 'trees' | 'conservation';
+}
+
+export default function ProjectSnippet({
+  project,
+  shouldDisplayPopup,
+}: Props): ReactElement {
+  const router = useRouter();
+  const tCommon = useTranslations('Common');
+  const { embed } = useContext(ParamsContext);
+
+  const progressPercentage = useMemo(() => {
+    if (project.purpose === 'trees' && project.countTarget) {
+      return Math.min((project.countPlanted / project.countTarget) * 100, 100);
+    }
+    return 0;
+  }, [(project as TreeProjectConcise).countPlanted, project.countTarget]);
+
+  const ecosystem =
+    project._scope === 'map' ? project.ecosystem : project.metadata.ecosystem;
+  const _isTopProject = (project as TreeProjectConcise).isTopProject;
+  const _isApproved = (project as TreeProjectConcise).isApproved;
+
+  const handleClick = () => {
+    if (embed === 'true') {
+      window.open(`/t/${project.tpo.slug}`, '_top');
+    } else {
+      router.push(`/t/${project.tpo.slug}`);
+    }
+  };
+
+  const getBackgroundClass = () => {
+    if (!project.allowDonations) return `${style.noDonation}`;
+    if (_isTopProject && _isApproved) return `${style.tpoBackground}`;
+    return '';
+  };
```

Comment:
1. `getBackgroundClass` is not a very informative name.
2. `getBackgroundClass` gets called for each render, for each project in the list. It might make sense to memoize the value.